src/main/java/edu/kis/powp/jobs2d/command/ComplexCommand.java

```diff
@@ -19,13 +19,18 @@ public void addCommand(DriverCommand command) {
 
     @Override
     public void execute(Job2dDriver driver){
-        while (this.iterator().hasNext()){
-            this.iterator().next().execute(driver);
+        for (DriverCommand command : commandList) {
+            command.execute(driver);
         }
     }
 
     @Override
     public Iterator<DriverCommand> iterator() {
         return commandList.iterator();
     }
-}
+
+    @Override
+    public int accept(DriverCommandVisitor visitor) {
+        return visitor.visit(this);
+    }
```

Comment:
I think this method can be moved to the ICompoundCommand interface (as default method)